.github/report-flaky-tests/index.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' ).promises;
+const path = require( 'path' );
+const github = require( '@actions/github' );
+const core = require( '@actions/core' );
+const unzipper = require( 'unzipper' );
+const { formatResultsErrors } = require( 'jest-message-util' );
+
+const TEST_RESULTS_LIST = {
+	open: `<!-- __TEST_RESULTS_LIST__ -->`,
+	close: `<!-- /__TEST_RESULTS_LIST__ -->`,
+};
+const TEST_RESULT = {
+	open: '<!-- __TEST_RESULT__ -->',
+	close: '<!-- /__TEST_RESULT__ -->',
+};
+
+const metaData = {
+	render: ( json ) => `<!-- __META_DATA__:${ JSON.stringify( json ) } -->`,
+	get: ( str ) => {
+		const matched = str.match( /<!-- __META_DATA__:(.*) -->/ );
+		if ( matched ) {
+			return JSON.parse( matched[ 1 ] );
+		}
+	},
+};
+
+( async function run() {
+	const token = core.getInput( 'repo-token', { required: true } );
+	const label = core.getInput( 'label', { required: true } );
+	const artifactName = core.getInput( 'artifact-name', { required: true } );
+
+	const octokit = github.getOctokit( token );
+
+	const flakyTests = await downloadReportFromArtifact(
+		octokit,
+		artifactName
+	);
+
+	if ( ! flakyTests ) {
+		return;
+	}
+
+	const { data: issues } = await octokit.rest.issues.listForRepo( {
+		...github.context.repo,
+		labels: label,
+		per_page: 100,
+	} );
+
+	for ( const flakyTestKey in flakyTests ) {
+		const [ testPath, testTitle ] = getTestPathAndTitle( flakyTestKey );
+		const issueTitle = getIssueTitle( testTitle );
+		const reportedIssue = issues.find(
+			( issue ) => issue.title === issueTitle
```

Comment:
I wonder if we should use a `key` that we store in the issue's metadata? Otherwise I fear somebody may accidentally rename the issue causing duplicates.

Reply:
Hmm, I don't understand what you mean here? How do generate such key? How do we map a test to the key?

Reply:
You could make a key from the title itself and then instead of saerching by title search in body for key. Key can be any hash from the title string. But I don't know about searching in body for string.

Reply:
Essentially something like:
```
let title = "Flaky test ...";
const hash = md5(title).substr(0,6);
title = `${title} (${hash})`;
`

Reply:
I thought that we had a natural key for each failing test. Re-reading this again, we don't... never mind!